Compile error with GCC 15

I'm having issues getting ringing-lib to compile with GCC 15:

```
methodset.cpp: In member function ‘size_t ringing::methodset::import_library(const std::string&)’:
methodset.cpp:211:16: error: call of overloaded ‘append(ringing::library::const_iterator, ringing::library::const_iterator)’ is ambiguous
  211 |   return append( l.begin(), l.end() );
      |          ~~~~~~^~~~~~~~~~~~~~~~~~~~~~
methodset.cpp:211:16: note: there are 4 candidates
In file included from ../ringing/methodset.h:34:
../ringing/libout.h:48:8: note: candidate 1: ‘void ringing::libout::append(Iterator, Iterator) [with Iterator = ringing::library_base::const_iterator]’
   48 |   void append( Iterator first, Iterator last ) {
      |        ^~~~~~
../ringing/methodset.h:77:10: note: candidate 2: ‘size_t ringing::methodset::append(InputIterator, InputIterator) [with InputIterator = ringing::library_base::const_iterator; size_t = long unsigned int]’
   77 |   size_t append( InputIterator first, InputIterator last ) {
      |          ^~~~~~
```

(despite GCC claiming there are 4 candidates, it only prints 2; see the full output below)

There are also a lot of warnings about a "different exception specifier":

```
In file included from ../ringing/common.h:35,
                 from proof.cpp:23:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/stdexcept:40,
                 from proof.cpp:29:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
```

I'm not a C++ expert by any means, so I have no idea how to fix this!

<details><summary>Full compile output</summary>

```
[josh@aldebaran:~/c/ringing-lib]$ gcc --version
gcc (GCC) 15.2.0
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[josh@aldebaran:~/c/ringing-lib]$ make
for dir in ringing utils apps ; do \
  make -C $dir all || exit; \
done
make[1]: Entering directory '/home/josh/c/ringing-lib/ringing'
for lib in core main; do \
  make -f $lib.mk all; \
done
make[2]: Entering directory '/home/josh/c/ringing-lib/ringing'
g++ -I.. -MM -MT .objs/bell.o bell.cpp -MF .deps/bell.d
g++ -I.. -MM -MT .objs/change.o change.cpp -MF .deps/change.d
g++ -I.. -MM -MT .objs/row.o row.cpp -MF .deps/row.d
g++ -I.. -MM -MT .objs/mathutils.o mathutils.cpp -MF .deps/mathutils.d
g++ -I.. -MM -MT .objs/place_notation.o place_notation.cpp -MF .deps/place_notation.d
g++ -I.. -MM -MT .objs/method.o method.cpp -MF .deps/method.d
g++ -I.. -MM -MT .objs/methodset.o methodset.cpp -MF .deps/methodset.d
g++ -I.. -MM -MT .objs/library.o library.cpp -MF .deps/library.d
g++ -I.. -MM -MT .objs/method_stream.o method_stream.cpp -MF .deps/method_stream.d
g++ -I.. -MM -MT .objs/libfacet.o libfacet.cpp -MF .deps/libfacet.d
g++ -I.. -MM -MT .objs/libout.o libout.cpp -MF .deps/libout.d
g++ -I.. -MM -MT .objs/litelib.o litelib.cpp -MF .deps/litelib.d
g++ -I.. -MM -MT .objs/xmllib.o xmllib.cpp -MF .deps/xmllib.d
g++ -I.. -MM -MT .objs/xmlout.o xmlout.cpp -MF .deps/xmlout.d
g++ -I.. -MM -MT .objs/peal.o peal.cpp -MF .deps/peal.d
g++ -I.. -MM -MT .objs/lexical_cast.o lexical_cast.cpp -MF .deps/lexical_cast.d
g++ -I.. -MM -MT .objs/stl.o stl.cpp -MF .deps/stl.d
g++ -I.. -MM -MT .objs/dom_stub.o dom_stub.cpp -MF .deps/dom_stub.d
g++ -I.. -g -O2 -fPIC -c -o .objs/bell.o bell.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/change.o change.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/row.o row.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/mathutils.o mathutils.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/place_notation.o place_notation.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/method.o method.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/methodset.o methodset.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/method_stream.o method_stream.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/library.o library.cpp
In file included from ../ringing/common.h:35,
                 from place_notation.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/algorithm:62,
                 from ../ringing/place_notation.h:38,
                 from place_notation.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from change.cpp:22:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/vector:64,
                 from change.cpp:29:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from library.cpp:22:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/list:64,
                 from ../ringing/library.h:38,
                 from library.cpp:28:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from method_stream.cpp:18:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/ostream.h:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ostream:42,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/iostream:43,
                 from method_stream.cpp:24:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from methodset.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/list:64,
                 from ../ringing/library.h:38,
                 from ../ringing/methodset.h:33,
                 from methodset.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
methodset.cpp: In member function ‘size_t ringing::methodset::import_library(const std::string&)’:
methodset.cpp:211:16: error: call of overloaded ‘append(ringing::library::const_iterator, ringing::library::const_iterator)’ is ambiguous
  211 |   return append( l.begin(), l.end() );
      |          ~~~~~~^~~~~~~~~~~~~~~~~~~~~~
methodset.cpp:211:16: note: there are 4 candidates
In file included from ../ringing/methodset.h:34:
../ringing/libout.h:48:8: note: candidate 1: ‘void ringing::libout::append(Iterator, Iterator) [with Iterator = ringing::library_base::const_iterator]’
   48 |   void append( Iterator first, Iterator last ) {
      |        ^~~~~~
../ringing/methodset.h:77:10: note: candidate 2: ‘size_t ringing::methodset::append(InputIterator, InputIterator) [with InputIterator = ringing::library_base::const_iterator; size_t = long unsigned int]’
   77 |   size_t append( InputIterator first, InputIterator last ) {
      |          ^~~~~~
g++ -I.. -g -O2 -fPIC -c -o .objs/libfacet.o libfacet.cpp
In file included from ../ringing/common.h:35,
                 from row.cpp:22:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_iterator.h:66,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/iterator:65,
                 from row.cpp:42:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
g++ -I.. -g -O2 -fPIC -c -o .objs/libout.o libout.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/litelib.o litelib.cpp
In file included from ../ringing/common.h:35,
                 from method.cpp:22:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/ostream.h:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ostream:42,
                 from ../ringing/change.h:34,
                 from method.cpp:41:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
make[2]: *** [../build/compile.mk:17: .objs/methodset.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from ../ringing/common.h:35,
                 from litelib.cpp:19:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/istream:42,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/fstream:42,
                 from litelib.cpp:30:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from libout.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/new_allocator.h:36,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/x86_64-unknown-linux-gnu/bits/c++allocator.h:33,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/allocator.h:46,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/string:45,
                 from ../ringing/libbase.h:31,
                 from ../ringing/libout.h:32,
                 from libout.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
make[2]: Leaving directory '/home/josh/c/ringing-lib/ringing'
make[2]: Entering directory '/home/josh/c/ringing-lib/ringing'
g++ -I.. -MM -MT .objs/mslib.o mslib.cpp -MF .deps/mslib.d
g++ -I.. -MM -MT .objs/cclib.o cclib.cpp -MF .deps/cclib.d
g++ -I.. -MM -MT .objs/extent.o extent.cpp -MF .deps/extent.d
g++ -I.. -MM -MT .objs/group.o group.cpp -MF .deps/group.d
g++ -I.. -MM -MT .objs/proof.o proof.cpp -MF .deps/proof.d
g++ -I.. -MM -MT .objs/falseness.o falseness.cpp -MF .deps/falseness.d
g++ -I.. -MM -MT .objs/row_wildcard.o row_wildcard.cpp -MF .deps/row_wildcard.d
g++ -I.. -MM -MT .objs/touch.o touch.cpp -MF .deps/touch.d
g++ -I.. -MM -MT .objs/music.o music.cpp -MF .deps/music.d
g++ -I.. -MM -MT .objs/print.o print.cpp -MF .deps/print.d
g++ -I.. -MM -MT .objs/print_ps.o print_ps.cpp -MF .deps/print_ps.d
g++ -I.. -MM -MT .objs/dimension.o dimension.cpp -MF .deps/dimension.d
g++ -I.. -MM -MT .objs/printm.o printm.cpp -MF .deps/printm.d
g++ -I.. -MM -MT .objs/print_pdf.o print_pdf.cpp -MF .deps/print_pdf.d
g++ -I.. -MM -MT .objs/pdf_fonts.o pdf_fonts.cpp -MF .deps/pdf_fonts.d
g++ -I.. -MM -MT .objs/search_base.o search_base.cpp -MF .deps/search_base.d
g++ -I.. -MM -MT .objs/basic_search.o basic_search.cpp -MF .deps/basic_search.d
g++ -I.. -MM -MT .objs/multtab.o multtab.cpp -MF .deps/multtab.d
g++ -I.. -MM -MT .objs/table_search.o table_search.cpp -MF .deps/table_search.d
g++ -I.. -MM -MT .objs/streamutils.o streamutils.cpp -MF .deps/streamutils.d
g++ -I.. -g -O2 -fPIC -c -o .objs/mslib.o mslib.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/cclib.o cclib.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/extent.o extent.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/methodset.o methodset.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/group.o group.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/proof.o proof.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/falseness.o falseness.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/touch.o touch.cpp
g++ -I.. -g -O2 -fPIC -c -o .objs/row_wildcard.o row_wildcard.cpp
In file included from ../ringing/common.h:35,
                 from proof.cpp:23:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/stdexcept:40,
                 from proof.cpp:29:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from extent.cpp:21:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/algorithm:62,
                 from extent.cpp:35:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from falseness.cpp:21:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/ostream.h:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ostream:42,
                 from ../ringing/row.h:42,
                 from falseness.cpp:27:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::falseness_table; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/falseness.h:146:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  146 | RINGING_DELEGATE_STD_SWAP( falseness_table )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::falseness_table; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::false_courses; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/falseness.h:147:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  147 | RINGING_DELEGATE_STD_SWAP( false_courses )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::false_courses; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from mslib.cpp:21:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/list:64,
                 from ../ringing/library.h:38,
                 from ../ringing/mslib.h:34,
                 from mslib.cpp:27:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from group.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/ostream.h:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ostream:42,
                 from ../ringing/row.h:42,
                 from group.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::group; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/group.h:123:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  123 | RINGING_DELEGATE_STD_SWAP( group )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::group; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from row_wildcard.cpp:21:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/stdexcept:40,
                 from ../ringing/row_wildcard.h:37,
                 from row_wildcard.cpp:27:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::group; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/group.h:123:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  123 | RINGING_DELEGATE_STD_SWAP( group )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::group; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from cclib.cpp:22:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/list:64,
                 from ../ringing/library.h:38,
                 from ../ringing/cclib.h:34,
                 from cclib.cpp:28:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from methodset.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/list:64,
                 from ../ringing/library.h:38,
                 from ../ringing/methodset.h:33,
                 from methodset.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
In file included from ../ringing/common.h:35,
                 from touch.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_pair.h:61,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_algobase.h:64,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/algorithm:62,
                 from touch.cpp:29:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
methodset.cpp: In member function ‘size_t ringing::methodset::import_library(const std::string&)’:
methodset.cpp:211:16: error: call of overloaded ‘append(ringing::library::const_iterator, ringing::library::const_iterator)’ is ambiguous
  211 |   return append( l.begin(), l.end() );
      |          ~~~~~~^~~~~~~~~~~~~~~~~~~~~~
methodset.cpp:211:16: note: there are 4 candidates
In file included from ../ringing/methodset.h:34:
../ringing/libout.h:48:8: note: candidate 1: ‘void ringing::libout::append(Iterator, Iterator) [with Iterator = ringing::library_base::const_iterator]’
   48 |   void append( Iterator first, Iterator last ) {
      |        ^~~~~~
../ringing/methodset.h:77:10: note: candidate 2: ‘size_t ringing::methodset::append(InputIterator, InputIterator) [with InputIterator = ringing::library_base::const_iterator; size_t = long unsigned int]’
   77 |   size_t append( InputIterator first, InputIterator last ) {
      |          ^~~~~~
g++ -I.. -g -O2 -fPIC -c -o .objs/music.o music.cpp
make[2]: *** [../build/compile.mk:17: .objs/methodset.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from ../ringing/common.h:35,
                 from music.cpp:20:
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/change.h:141:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  141 | RINGING_DELEGATE_STD_SWAP( change )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/exception_ptr.h:41,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/exception:168,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ios:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/ostream.h:43,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/ostream:42,
                 from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/iostream:43,
                 from ../ringing/music.h:42,
                 from music.cpp:26:
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::change; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
../ringing/macros.h:172:8: warning: declaration of ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’ has a different exception specifier
  172 |   void swap<RINGING_PREFIX T>(RINGING_PREFIX T &a, RINGING_PREFIX T &b) \
      |        ^~~~~~~~~~~~~~~~~~~~~~
../ringing/row.h:269:1: note: in expansion of macro ‘RINGING_DELEGATE_STD_SWAP’
  269 | RINGING_DELEGATE_STD_SWAP( row )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/move.h:227:5: note: from previous declaration ‘std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) noexcept [with _Tp = ringing::row; _Require<__not_<__is_tuple_like<_Tp> >, is_move_constructible<_Tp>, is_move_assignable<_Tp> > = void]’
  227 |     swap(_Tp& __a, _Tp& __b)
      |     ^~~~
make[2]: Leaving directory '/home/josh/c/ringing-lib/ringing'
make[1]: *** [Makefile:5: all] Error 2
make[1]: Leaving directory '/home/josh/c/ringing-lib/ringing'
make: *** [Makefile:4: all] Error 2
```
</details>

--- a/ringing/methodset.cpp
+++ b/ringing/methodset.cpp
@@ -191,6 +191,11 @@ size_t methodset::size() const
   return this->libout::get_impl<impl>()->size();
 }
 
+void methodset::append( library_entry const& entry )
+{
+  this->libout::append(entry);
+}
+
 void methodset::append( method const& m )
 {
   this->libout::get_impl<impl>()->append( m );

--- a/ringing/methodset.h
+++ b/ringing/methodset.h
@@ -70,7 +70,7 @@ public:
   size_t size() const;
 
   // Currently libout doesn't have an append that takes a single method
-  using libout::append;
+  void append( library_entry const& entry );
   void append( method const& );
 
   template <class InputIterator>

